src/cmd/helm.rs

```diff
@@ -506,7 +521,16 @@ impl Helm {
 
         match helm_ret {
             // Ok is ok
-            Ok(_) => Ok(()),
+            Ok(_) => {
+                if chart.backup_resources.is_some() {
+                    if let Err(e) =
+                        apply_chart_backup(&self.kubernetes_config, &root_dir_path, &self.get_all_envs(envs), chart)
```

Comment:
What happens to the backup if the upgrade fails ?

Reply:
nothing, it should be deleted I think

Reply:
where ?